in-vehicle-digital-twin/src/digitaltwin_impl.rs

```diff
@@ -34,36 +32,23 @@ impl DigitalTwin for DigitalTwinImpl {
         request: Request<FindByIdRequest>,
     ) -> Result<Response<FindByIdResponse>, Status> {
         let request_inner = request.into_inner();
-        let entity_id = request_inner.entity_id;
+
+        let entity_id = request_inner.id;
 
         info!("Received a find_by_id request for entity id {entity_id}");
 
-        let dtdl;
+        let entity_access_info;
 
         // This block controls the lifetime of the lock.
         {
-            let lock: RwLockReadGuard<HashMap<String, Value>> = self.entity_map.read();
-            let val_option = lock.get(&entity_id);
-            let val = match val_option {
-                Some(v) => v,
-                None => {
-                    return Err(Status::not_found(format!(
-                        "Unable to find the DTDL for entity id {entity_id}"
-                    )))
-                }
-            };
-
-            dtdl = match serde_json::to_string_pretty(&val) {
-                Ok(content) => content,
-                Err(error) => {
-                    return Err(Status::internal(format!(
-                        "Unexpected error with the DTDL for entity id {entity_id}: {error:?}"
-                    )))
-                }
-            };
+            let lock: RwLockReadGuard<HashMap<String, EntityAccessInfo>> =
```

Comment:
is the type annotation necessary here? it's very verbose and looks messy, I'd suggest removing it if possible

Reply:
This is a stylistic preference.  Leaving as is.